pkg/kubectl/explain/explain.go

```diff
@@ -30,6 +30,10 @@ type fieldsPrinter interface {
 
 func splitDotNotation(model string) (string, []string) {
 	var fieldsPath []string
+
+	// ignore trailing period
+	model = strings.TrimSuffix(model, ".")
+
```

Comment:
I tested the change and it works as such but as @ahmetb said, definitely adding a test is desired. On a side note, it may be a good idea to provide better error message than ignoring trailing period, as it can be an user error that some string value was missed to be specified after tailing period by user. So instead of ignoring trailing period, provide a better error message, for example, something like field "ingress.spec.rules.http.paths." doesn't exist instead of field "" does not exist. As @ahmetb mentioned the proposal of ignoring trailing period may need more discussion but adding a better error should be quick and safer. Thanks!